src/github/copilotRemoteAgent.ts

```diff
@@ -76,6 +100,61 @@ export class CopilotRemoteAgentManager extends Disposable {
 			.getConfiguration(CODING_AGENT).get(CODING_AGENT_ENABLED, false);
 	}
 
+	async isAssignable(): Promise<boolean> {
+		const repoInfo = await this.repoInfo();
+		if (!repoInfo) {
+			return false;
+		}
+
+		const { owner, repo } = repoInfo;
+		const folderManager = this.getFolderManagerForRepo(owner, repo);
+
+		try {
+			// Ensure assignable users are loaded
+			await folderManager.getAssignableUsers();
+			const allAssignableUsers = folderManager.getAllAssignableUsers();
+
+			if (!allAssignableUsers) {
+				return false;
+			}
+
+			// Check if any of the copilot logins are in the assignable users
+			return allAssignableUsers.some(user => COPILOT_LOGINS.includes(user.login));
+		} catch (error) {
+			// If there's an error fetching assignable users, assume not assignable
+			return false;
+		}
+	}
+
+	async isAvailable(): Promise<boolean> {
+		// Check if the manager is enabled, copilot API is available, and it's assignable
+		if (!this.enabled()) {
+			return false;
+		}
+
+		const repoInfo = await this.repoInfo();
+		if (!repoInfo) {
+			return false;
+		}
+
+		const copilotApi = await this.copilotApi;
+		if (!copilotApi) {
+			return false;
+		}
+
+		return await this.isAssignable();
+	}
+
+	private async updateAssignabilityContext(): Promise<void> {
+		try {
+			const available = await this.isAvailable();
+			commands.setContext('copilotCodingAgentAssignable', available);
```

Comment:
I try to keep the contexts here:

https://github.com/microsoft/vscode-pull-request-github/blob/e7b8763aa90837ac293cb91bfb26ae4dfda8af1e/src/common/executeCommands.ts#L8-L9